packages/core/components/Puck/components/Canvas/index.tsx

```diff
@@ -80,26 +90,25 @@ export const Canvas = () => {
     return { width: 0, height: 0 };
   }, [frameRef]);
 
-  const resetAutoZoom = useCallback(
-    (newViewports: UiState["viewports"] = viewports) => {
-      if (frameRef.current) {
-        setZoomConfig(
-          getZoomConfig(
-            newViewports?.current,
-            frameRef.current,
-            zoomConfig.zoom
-          )
-        );
-      }
-    },
-    [frameRef, zoomConfig, viewports]
-  );
-
   // Auto zoom
   useEffect(() => {
-    setShowTransition(false);
-    resetAutoZoom(viewports);
-  }, [frameRef, leftSideBarVisible, rightSideBarVisible]);
+    if (!isResettingZoomRef.current) {
+      isResettingZoomRef.current = true;
```

Comment:
I don't quite understand what this code is doing and, if it's realy necessary, why it couldn't be moved within `useResetAutoZoom` rather than being repeated throughout this component

Reply:
@tlahmann please let me know about this one! I'd like to get this PR merged this week for our 0.20 release

Reply:
Hey @chrisvxd. you're absolutely right, the hook can also handle the reference and other options. I have reworked that.
While working on this i had minor issues with the reference object preventing the reset calling itself